### ISSUE TITLE
Human two populaiton test

### ISSUE THREAD

**Issue by kmanoharan01** [gphocs-dev/G-PhoCS#82]
Hi

I am getting the following error:

####
Error: in recorded genealogy log likelihood for gen 27: (recorded = 90.3186, actual = 90.3186, diff = 1.22089e-05, 1-ratio = -1.35175e-07).

Error:  --  Aborting when logging after MCMC iteration 11149, due to data structure inconsistency.

Please find below log file. Could you please tell me what I am doing wrong here?

thanks,

#######

GENERAL-INFO-START

	seq-file           hg19_chr1_neutral.txt
	trace-file          hg19_chr1_neutral.log				
	locus-mut-rate          CONST

	mcmc-iterations	  100000
	iterations-per-log  50
	logs-per-line       10

	find-finetunes		FALSE
	finetune-coal-time	0.01		
	finetune-mig-time	0.3		
	finetune-theta		0.04
	finetune-mig-rate	0.02
	finetune-tau		0.0000008
	finetune-mixing		0.003
#   finetune-locus-rate 0.3
	
	tau-theta-print		10000.0
	tau-theta-alpha		1.0			# for STD/mean ratio of 100%
	tau-theta-beta		10000.0		# for mean of 1e-4

	mig-rate-print		0.001
	mig-rate-alpha		0.002
	mig-rate-beta		0.00001

GENERAL-INFO-END

CURRENT-POPS-START	

	POP-START
		name		A
		samples		sample1 d
	POP-END

	POP-START
		name		B
		samples		sample2 d
	POP-END

CURRENT-POPS-END

ANCESTRAL-POPS-START

	POP-START
		name			AB
		children		A		B
		tau-initial	0.000005
		tau-beta		20000.0	
		finetune-tau			0.0000008
	POP-END

ANCESTRAL-POPS-END

MIG-BANDS-START	
	BAND-START		
       source A 
       target  B
       mig-rate-print 0.1
	BAND-END

MIG-BANDS-END
#####

**Comment by igronau** [comment]
Sorry for the delayed response. The error means that the sampler noticed an accumulation of rounding errors in its internal computations. This can happen when you have very small time intervals between events (coalescence, migration, divergence times). This will happen when certain parameters get driven to the boundary of their valid space (very small \theta or \tau or very large m). You can look at the parameter trace file to see what the current parameter values are to get a hint.

I can tell you that in some data sets, it's very hard to differentiate migration between sister populations (A and B in your case) versus very recent divergence. This can drive the sampler to the edge of the parameter space and cause this kind of unexpected behavior.

**Comment by kmanoharan01** [comment]
Thank you for your reply.

In that case, may I know what is the minimum age of divergence time G-PhoCS can detect with high confident?

**Comment by igronau** [comment]
It depends on the scenario (number of loci, migration rate, etc.). I would run a version without any gene flow (no migration bands) as some sort of control. Then, you can run a version with gene flow, and try both direction. There is some trial and error often involved here until you get the settings right for your data set.

**Comment by kmanoharan01** [comment]
Thank you, I performed as you suggested without gene flow using and still got the error; only when I changed to tau-theta-print parameter value (1000) it runs without any errors: Could you help me to understand about this parameter:

tau-theta-print		10000.0 # Is it starting time (10,000 years ago)?
tau-theta-alpha		1.0			# for STD/mean ratio of 100%
tau-theta-beta		10000.0		# for mean of 1e-4

###
tau-initial	0.000005 
tau-beta		20000.0	(Is this the end time of the model?)
finetune-tau			0.0000008

Here 10000 represent number of years? 

Basically, I'm trying to covert tau/G-Phoc result to number of year (eg; divergence time);

How could I do this?

Thanks,
Mano

**Comment by igronau** [comment]
You should read the manual, since it explain all of this, and probably some of the issues having to do with conversion of estimates to years. Another good resource is the supplement of our paper from 2011, where we introduce G-PhoCS. 

**tau-theta-print** is just a constant used for factoring the printed values of tau and theta. I find it odd that changing it will make any difference in the sampling process. So, i suspect it is some other change you made that made the program work without fault.

 **tau-beta** is the beta parameter of the Gamma distribution used to define the priors for this tau parameter.

You'll find more valuable info in the manual.

**Comment by kmanoharan01** [comment]
Yes, Thank you very much for your advise.
